packages/ui/src/multiplying-architecture/KaotoEditorChannelApi.ts

```diff
@@ -12,4 +12,32 @@ export interface KaotoEditorChannelApi extends KogitoEditorChannelApi {
    * Returns the Kaoto VSCode settings defined.
    */
   getVSCodeKaotoSettings(): Promise<ISettingsModel>;
+
+  /**
+   * Get metadata querying a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to retrieve the preferences
+   */
+  getMetadata<T>(path: string, key: string): Promise<T>;
+
+  /**
+   * Save preferences to a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to set the preferences
+   * @param metadata The preferences to be saved
+   */
+  setMetadata<T>(path: string, key: string, metadata: T): Promise<void>;
+
+  /**
+   * Retrieve resource content
+   * @param path The path of the resource, relative to the file being edited
+   */
+  getResourceContent(path: string): Promise<string>;
```

Comment:
shouldn't we ba bel to return undefined or an exception in case the file doesn't exist?

Reply:
Good point, initially I thought about adding an exception, but since we'll be reading files in text mode, probably `undefined` would be enough. I'll update the API.